doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,368 @@
+Moving to Highway is likely to be a significant development effort.
+Longer-term, this will hopefully be offset by Highway itself having more
+maintainer bandwidth to deal with ongoing issues in compiler support and adding
+new platforms. 
+
+Highway being used by other projects, like Chromium and `JPEG XL`_ (see
+`this more complete list <https://google.github.io/highway/en/master/README.html#examples>`__
+in the Highway documentation), does imply that there is likely to be a benefit
+of a wider range of testing and bug reporting/fixing.
+
+One concern is that new instructions may have to be added, and that that is
+often best done as part of the process of developing the numerical kernel that
+needs the instruction. This will be a little more clumsy if the instruction
+lives in Highway which is a git submodule inside the NumPy repo - there will be
+a need to implement a temporary/generic version first, and then update the
+submodule after upstreaming the new intrinsic.
+
+Documentation-wise, Highway would be a clear win. NumPy's
+`CPU/SIMD Optimizations`_ docs are fairly sparse compared to
+`the Highway docs`_.
+
+Migration strategy - can it be gradual?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    
+This is a story of two halves. Moving to Highway's statically dispatched
+intrinsics could be done gradually, as already seen in PR `gh-24018`_. However,
+adopting Highway's way of performing runtime dispatching has to be done in one
+go - we can't (or shouldn't) have two ways of doing that.
+
+
+Highway policies for compiler and platform support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    
+When adding new instructions, Highway has a policy that they must be
+implemented in a way that fairly balances across CPU architectures.
+
+Regarding the support status and whether all currently-supported architectures
+will remain supported, Jan stated that Highway can commit to the following:
+
+1. If it cross-compiles with Clang and can be tested via standard QEMU, it can
+   go into Highway's CI.
+2. If it cross-compiles via clang/gcc and can be tested with a new QEMU
+   (possibly with extra flags), then it can be support via manual testing
+   before each Highway release.
+3. Existing targets will remain supported as long as they compile/run in QEMU.
+
+Highway is not subject to Google's "no longer supported" strategy (or, as
+written in its README, *This is not an officially supported Google product*).
+That is not a bad thing; it means that it is less likely to go unsupported due
+to a Google business decision about the project. Quite a few well-known open
+source projects under the ``google`` GitHub org state this, e.g. `JAX`_ and
+`tcmalloc`_.
+
+
+Supported features/targets
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Both frameworks support a large set of platforms and SIMD instruction sets,
+as well as generic scalar/fallback versions. The main differences right now are:
+
+- NumPy supports IBM Z-system (s390x, VX/VXE/VXE2) while Highway does not,
```

Comment:
A Highway contributor has just added support for Z14 and Z15.

Reply:
Noted.